### DISCUSSION TITLE
Is it possible to horizontally scale multiple brokers?

### DISCUSSION THREAD

**Discussion by iiliev2** [moquette-io/moquette#711] [Q&A] [answered]
Is it possible to create a cluster of brokers which talk to each other with the goal of handling increasing load?

**Comment by enozcan** [comment]
I couldn't see clustering support in the codebase and am also wondering if that's possible via using a shared persistence layer - say an in-memory/disk-based implementation. But in this case, handling race between brokers would probably require additional care - e.g. one instance handling unsubscribe where another handles publish to this particular client - in this scenario publisher might stuck in the retry loop for this no-longer-open subscription. Or, say instance1 has the open connection for client1, then what will be the status for instance2 that handles publish to a topic for an active subscription - which is kept open in the instance1.

**Comment by iiliev2** [comment, marked as answer]
Found the answer, it is not yet possible, with intention to add that functionality: https://github.com/moquette-io/moquette/issues/131
